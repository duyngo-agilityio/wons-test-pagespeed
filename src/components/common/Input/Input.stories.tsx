import type { Meta, StoryObj } from '@storybook/react';

// components
import Input from './index';

const meta: Meta<typeof Input> = {
  title: 'Components/Common/Input',
  component: Input,
  parameters: {
    controls: { expanded: true },
  },

  args: {
    placeholder: 'Enter text...',
  },

  argTypes: {
    color: {
      description: 'Change the color of the input',
      options: ['default', 'primary', 'secondary'],
      control: { type: 'select' },
    },
    size: {
      description: 'Change the size of the input',
      options: ['xs', 'sm', 'md', 'lg', 'xl'],
      control: { type: 'select' },
    },
    placeholder: {
      description: 'Input placeholder text',
      control: { type: 'text' },
    },
    disabled: {
      description: 'Disable the input field',
      control: { type: 'boolean' },
    },
    required: {
      description: 'Set the input field as required',
      control: { type: 'boolean' },
    },
    radius: {
      description: 'Set the border radius of the input',
      options: ['none', 'sm', 'md', 'lg', 'xl'],
      control: { type: 'select' },
    },
    border: {
      description: 'Set the border style of the input',
    },
  },
};

export default meta;
type Story = StoryObj<typeof Input>;

export const Default: Story = {
  args: {},
};
