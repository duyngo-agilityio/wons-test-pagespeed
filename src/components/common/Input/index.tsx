'use client';

// Libs
import {
  extendVariants,
  InputProps,
  Input as NextUIInput,
} from '@nextui-org/react';

const CustomInput = extendVariants(NextUIInput, {
  variants: {
    color: {
      primary: {
        inputWrapper: [
          'bg-gray-50 dark:bg-gray-600',
          'hover:!bg-gray-200/50 dark:hover:!bg-gray-900',
          'focus-within:!bg-gray-50 dark:focus-within:bg-gray-600',
          'group-data-[focus=true]:!bg-gray-50 dark:group-data-[focus=true]:!bg-gray-600',
        ],
        input: [
          '!text-blue-800/70 dark:!text-white/70',
          'placeholder:text-blue-800/40 dark:placeholder:text-white/40',
          'bg-transparent',
        ],
        label: [
          'group-data-[filled-within=true]:!text-blue-800 dark:group-data-[filled-within=true]:!text-white',
        ],
      },
      secondary: {
        inputWrapper: [
          'bg-gray-200/30 dark:bg-gray-600/30',
          'hover:!bg-gray-200/50 dark:hover:!bg-gray-600/80',
          'focus-within:!bg-gray-200/30 dark:focus-within:bg-gray-600/30',
          'group-data-[focus=true]:!bg-gray-200/30 dark:group-data-[focus=true]:!bg-gray-600/30',
        ],
        input: [
          '!text-blue-800/70 dark:!text-white/70',
          'placeholder:text-blue-800/40 dark:placeholder:text-white/40',
        ],
        label:
          'group-data-[filled-within=true]:text-blue-800 dark:group-data-[filled-within=true]:text-white',
      },
    },
    size: {
      sm: {
        inputWrapper: 'h-[42px] px-5 pt-3 pb-3.75',
        input: 'text-xl',
      },
      md: {
        inputWrapper: 'h-12.5 p-3.75',
        input: 'text-md',
        base: 'data-[has-label=true]:mt-[29px]',
        label: 'font-medium text-xl top-5',
      },
    },
    radius: {
      sm: {
        inputWrapper: 'rounded-[5px]',
      },
      md: {
        inputWrapper: 'rounded-[10px]',
      },
    },
  },

  defaultVariants: {
    color: 'primary',
    size: 'md',
    radius: 'md',
    labelPlacement: 'outside',
    placeholder: ' ',
  },
});

const Input = ({ ...props }: InputProps): JSX.Element => (
  <CustomInput
    classNames={{
      inputWrapper:
        'group-data-[focus-visible=true]:ring-2 group-data-[focus-visible=true]:ring-offset-2 group-data-[focus-visible=true]:ring-blue-500',
    }}
    {...props}
  />
);

export default Input;
