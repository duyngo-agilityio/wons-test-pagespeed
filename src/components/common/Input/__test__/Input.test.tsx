// Components
import Input from '../index';

describe('Input Component', () => {
  it('should match with snapshot', () => {
    const component = testLibJestUtils.render(
      <Input
        label="Input Label"
        name="inputName"
        labelPlacement="outside"
        placeholder="Input Placeholder"
      />,
    );
    expect(component).toMatchSnapshot();
  });
});
