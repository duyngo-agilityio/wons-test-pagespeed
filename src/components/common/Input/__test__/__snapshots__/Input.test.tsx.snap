// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Component should match with snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        data-overlay-container="true"
      >
        <script>
          !function(){try{var d=document.documentElement,c=d.classList;c.remove('light','dark');var e=localStorage.getItem('theme');if('system'===e||(!e&&false)){var t='(prefers-color-scheme: dark)',m=window.matchMedia(t);if(m.media!==t||m.matches){d.style.colorScheme = 'dark';c.add('dark')}else{d.style.colorScheme = 'light';c.add('light')}}else if(e){c.add(e|| '')}else{c.add('light')}if(e==='light'||e==='dark'||!e)d.style.colorScheme=e||'light'}catch(e){}}()
        </script>
         
        <div
          class="group flex flex-col data-[hidden=true]:hidden w-full group relative justify-end data-[has-label=true]:mt-[29px]"
          data-filled="true"
          data-filled-within="true"
          data-has-elements="true"
          data-has-label="true"
          data-slot="base"
        >
          <div
            class="h-full flex flex-col"
            data-slot="main-wrapper"
          >
            <div
              class="relative w-full inline-flex tap-highlight-transparent flex-row items-center shadow-sm gap-3 min-h-10 transition-background motion-reduce:transition-none !duration-150 data-[hover=true]:bg-primary-100 text-primary group-data-[focus=true]:bg-primary-50 placeholder:text-primary outline-none group-data-[focus-visible=true]:z-10 group-data-[focus-visible=true]:ring-offset-background bg-gray-50 dark:bg-gray-600 hover:!bg-gray-200/50 dark:hover:!bg-gray-900 focus-within:!bg-gray-50 dark:focus-within:bg-gray-600 group-data-[focus=true]:!bg-gray-50 dark:group-data-[focus=true]:!bg-gray-600 h-12.5 p-3.75 rounded-[10px] group-data-[focus-visible=true]:ring-2 group-data-[focus-visible=true]:ring-offset-2 group-data-[focus-visible=true]:ring-blue-500"
              data-slot="input-wrapper"
              style="cursor: text;"
            >
              <label
                class="absolute pointer-events-none origin-top-left rtl:origin-top-right subpixel-antialiased block will-change-auto !duration-200 !ease-out motion-reduce:transition-none transition-[transform,color,left,opacity] text-primary group-data-[filled-within=true]:pointer-events-auto pb-0 z-20 -translate-y-1/2 group-data-[filled-within=true]:left-0 left-3 rtl:left-auto rtl:right-3 group-data-[filled-within=true]:-translate-y-[calc(100%_+_theme(fontSize.small)/2_+_20px)] pe-2 max-w-full text-ellipsis overflow-hidden group-data-[filled-within=true]:!text-blue-800 dark:group-data-[filled-within=true]:!text-white font-medium text-xl top-5"
                data-slot="label"
                for="react-aria-:r0:"
                id="react-aria-:r1:"
              >
                Input Label
              </label>
              <div
                class="inline-flex w-full items-center h-full box-border"
                data-slot="inner-wrapper"
              >
                <input
                  aria-label="Input Label"
                  aria-labelledby="react-aria-:r0: react-aria-:r1:"
                  class="w-full font-normal !outline-none focus-visible:outline-none data-[has-start-content=true]:ps-1.5 data-[has-end-content=true]:pe-1.5 file:cursor-pointer file:bg-transparent file:border-0 autofill:bg-transparent bg-clip-text !text-blue-800/70 dark:!text-white/70 placeholder:text-blue-800/40 dark:placeholder:text-white/40 bg-transparent text-md"
                  data-slot="input"
                  id="react-aria-:r0:"
                  name="inputName"
                  placeholder="Input Placeholder"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="fixed top-4 right-4"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      data-overlay-container="true"
    >
      <script>
        !function(){try{var d=document.documentElement,c=d.classList;c.remove('light','dark');var e=localStorage.getItem('theme');if('system'===e||(!e&&false)){var t='(prefers-color-scheme: dark)',m=window.matchMedia(t);if(m.media!==t||m.matches){d.style.colorScheme = 'dark';c.add('dark')}else{d.style.colorScheme = 'light';c.add('light')}}else if(e){c.add(e|| '')}else{c.add('light')}if(e==='light'||e==='dark'||!e)d.style.colorScheme=e||'light'}catch(e){}}()
      </script>
       
      <div
        class="group flex flex-col data-[hidden=true]:hidden w-full group relative justify-end data-[has-label=true]:mt-[29px]"
        data-filled="true"
        data-filled-within="true"
        data-has-elements="true"
        data-has-label="true"
        data-slot="base"
      >
        <div
          class="h-full flex flex-col"
          data-slot="main-wrapper"
        >
          <div
            class="relative w-full inline-flex tap-highlight-transparent flex-row items-center shadow-sm gap-3 min-h-10 transition-background motion-reduce:transition-none !duration-150 data-[hover=true]:bg-primary-100 text-primary group-data-[focus=true]:bg-primary-50 placeholder:text-primary outline-none group-data-[focus-visible=true]:z-10 group-data-[focus-visible=true]:ring-offset-background bg-gray-50 dark:bg-gray-600 hover:!bg-gray-200/50 dark:hover:!bg-gray-900 focus-within:!bg-gray-50 dark:focus-within:bg-gray-600 group-data-[focus=true]:!bg-gray-50 dark:group-data-[focus=true]:!bg-gray-600 h-12.5 p-3.75 rounded-[10px] group-data-[focus-visible=true]:ring-2 group-data-[focus-visible=true]:ring-offset-2 group-data-[focus-visible=true]:ring-blue-500"
            data-slot="input-wrapper"
            style="cursor: text;"
          >
            <label
              class="absolute pointer-events-none origin-top-left rtl:origin-top-right subpixel-antialiased block will-change-auto !duration-200 !ease-out motion-reduce:transition-none transition-[transform,color,left,opacity] text-primary group-data-[filled-within=true]:pointer-events-auto pb-0 z-20 -translate-y-1/2 group-data-[filled-within=true]:left-0 left-3 rtl:left-auto rtl:right-3 group-data-[filled-within=true]:-translate-y-[calc(100%_+_theme(fontSize.small)/2_+_20px)] pe-2 max-w-full text-ellipsis overflow-hidden group-data-[filled-within=true]:!text-blue-800 dark:group-data-[filled-within=true]:!text-white font-medium text-xl top-5"
              data-slot="label"
              for="react-aria-:r0:"
              id="react-aria-:r1:"
            >
              Input Label
            </label>
            <div
              class="inline-flex w-full items-center h-full box-border"
              data-slot="inner-wrapper"
            >
              <input
                aria-label="Input Label"
                aria-labelledby="react-aria-:r0: react-aria-:r1:"
                class="w-full font-normal !outline-none focus-visible:outline-none data-[has-start-content=true]:ps-1.5 data-[has-end-content=true]:pe-1.5 file:cursor-pointer file:bg-transparent file:border-0 autofill:bg-transparent bg-clip-text !text-blue-800/70 dark:!text-white/70 placeholder:text-blue-800/40 dark:placeholder:text-white/40 bg-transparent text-md"
                data-slot="input"
                id="react-aria-:r0:"
                name="inputName"
                placeholder="Input Placeholder"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="fixed top-4 right-4"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
